src/app/_utils/PermissionGuard.ts

```diff
@@ -1,5 +1,5 @@
 import { Injectable } from '@angular/core';
-import { CanActivate, ActivatedRoute, Router, ActivatedRouteSnapshot } from '@angular/router';
+import { CanActivate, ActivatedRoute, Router, ActivatedRouteSnapshot, CanActivateChild } from '@angular/router';
 
 @Injectable({
     providedIn: "root",
```

Comment:
I have a small comment about DI concept here,
you provided guard in module AND and providedIn root.
Interestingly, but there will be two versions of guard in system :)

as for now - https://www.youtube.com/playlist?list=PLX7eV3JL9sfmJ6AaZj9eDlAKrJrEul4Vz 
give this playlist a try plus some resources

but we will talk about DI later a bit